Simple Incremental/Assets/Scripts/ObjectPooler.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/* This class is used for pooling objects and reusing objects for performance
+ * This works best when lots of the same object are spawned multiple times
+ * such as projectiles from a gun */
+public class ObjectPooler : MonoBehaviour
+{
+    public List<GameObject> pooledObjects;
+    public GameObject objectToPool;
+    public int amountToPool;
+
+    void Start()
+    {
+        pooledObjects = new List<GameObject>();
+        for (int i = 0; i < amountToPool; i++)
+        {
+            GameObject obj = (GameObject)Instantiate(objectToPool);
+            obj.transform.SetParent(transform);
+            obj.SetActive(false);
+            pooledObjects.Add(obj);
+        }
+    }
+
+    public GameObject GetPooledObject()
+    {
+        for (int i = 0; i < pooledObjects.Count; i++)
```

Comment:
Iterating through the object pool to find inactive objects is a waste of resources. This is the "old" way to do things. Take a look at the Queue data type!